### PR TITLE
chore: remove legacy glue table ods_events and remove ods_events and upsert users workflows

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
please add end-to-end test result for upgrading from v1.0.x to latest.

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 1]
```diff
@@ -326,9 +326,9 @@ function getUpdatableSql(sqlOrViewDefs: SQLDef[], oldSqlArray: string[], mustach
   const updateFilesInfo = [];
   const sqlStatements: string[] = [];
   for (const schemaOrViewDef of sqlOrViewDefs) {
-    logger.info(`schemaOrViewDef.updatable: ${schemaOrViewDef.updatable}`);
+    logger.info(`schemaOrViewDef: sqlFile: ${schemaOrViewDef.sqlFile} - updatable: ${schemaOrViewDef.updatable}`);
 
-    if (!oldSqlArray.includes(schemaOrViewDef.sqlFile)) {
+    if (! isSqlFileInOldArray(schemaOrViewDef.sqlFile, oldSqlArray)) {
```
Comment (zxkane) [inline src/analytics/lambdas/custom-resource/create-schemas.ts:331]: So will it exclude the view definition for `clickstream_lifecycle_daily_view`? However, the view is required for reporting 1.1.0 based on new `event` table.

@llmin should we drop the old view if it exists when upgrading from 1.0.x if you are using the same view name?

File: src/analytics/analytics-on-redshift.ts [hunk 2]
```diff
@@ -313,11 +295,10 @@ export class RedshiftAnalyticsStack extends NestedStack {
       createMetricsWidgetForRedshiftServerless(this, 'newServerless', {
         projectId: props.projectId,
         dataProcessingCronOrRateExpression: props.dataProcessingCronOrRateExpression,
-        upsertUsersCronOrRateExpression: props.upsertUsersWorkflowData.scheduleExpression,
+        scanMetadataCronOrRateExpression: props.dataProcessingCronOrRateExpression,
```
Comment (zxkane) [inline src/analytics/analytics-on-redshift.ts:298]: It's already removed in mainline.

File: src/analytics/analytics-on-redshift.ts [hunk 3]
```diff
@@ -328,11 +309,10 @@ export class RedshiftAnalyticsStack extends NestedStack {
       createMetricsWidgetForRedshiftServerless(this, 'existingServerless', {
         projectId: props.projectId,
         dataProcessingCronOrRateExpression: props.dataProcessingCronOrRateExpression,
-        upsertUsersCronOrRateExpression: props.upsertUsersWorkflowData.scheduleExpression,
+        scanMetadataCronOrRateExpression: props.dataProcessingCronOrRateExpression,
```
Comment (zxkane) [inline src/analytics/analytics-on-redshift.ts:312]: It's already removed in mainline.

File: src/analytics/analytics-on-redshift.ts [hunk 4]
```diff
@@ -342,10 +322,9 @@ export class RedshiftAnalyticsStack extends NestedStack {
       createMetricsWidgetForRedshiftCluster(this, {
         projectId: props.projectId,
         dataProcessingCronOrRateExpression: props.dataProcessingCronOrRateExpression,
-        upsertUsersCronOrRateExpression: props.upsertUsersWorkflowData.scheduleExpression,
+        scanMetadataCronOrRateExpression: props.dataProcessingCronOrRateExpression,
```
Comment (zxkane) [inline src/analytics/analytics-on-redshift.ts:325]: It's already removed in mainline.

File: src/analytics/analytics-on-redshift.ts [hunk 5]
```diff
@@ -393,8 +372,8 @@ function addCfnNag(stack: Stack) {
   addCfnNagToStack(stack, [
 
     ruleRolePolicyWithWildcardResources(
-      'UpsertUsersWorkflow/UpsertUsersStateMachine/Role/DefaultPolicy/Resource',
-      'UpsertUsersWorkflow', 'logs/xray'),
+      'ScanMetadataWorkflow/ScanMetadataStateMachine/Role/DefaultPolicy/Resource',
```
Comment (zxkane) [inline src/analytics/analytics-on-redshift.ts:375]: Why chaning to ScanMetadata? it's already in mainline

File: test/analytics/analytics-on-redshift/data-analytics-redshift-stack.test.ts [hunk 6]
```diff
@@ -3360,152 +3138,11 @@ describe('DataAnalyticsRedshiftStack lambda function test', () => {
     }
   });
 
-  test('Check UpsertUsersWorkflowUpsertUsersFn', () => {
+  test('Check ScanMetadataWorkflowCheckScanMetadataJobStatusFnSG', () => {
```
Comment (zxkane) [inline test/analytics/analytics-on-redshift/data-analytics-redshift-stack.test.ts:3141]: already having a test `'Check ScanMetadataWorkflowCheckScanMetadataStatusFnSG'`

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by zxkane** [comment]
@tyyzqmf please also remove `user table upsert frequency` in frontend.

Also, pay attention to update existing 1.0 pipeline.

**Comment by tyyzqmf** [comment]
> @tyyzqmf please also remove `user table upsert frequency` in frontend.
> 
> Also, pay attention to update existing 1.0 pipeline.

[delete redshift user table upsert frequency(frontend&backend)](https://github.com/awslabs/clickstream-analytics-on-aws/pull/330/commits/bec5bdab229f7ce6f86a704bc0eb35510ff4f842)

File: test/analytics/analytics-on-redshift/data-analytics-redshift-stack.test.ts (hunk 6 above)
Comment (amliuyong) [reply test/analytics/analytics-on-redshift/data-analytics-redshift-stack.test.ts:3141]: removed

File: src/analytics/analytics-on-redshift.ts (hunk 4 above)
Comment (amliuyong) [reply src/analytics/analytics-on-redshift.ts]: removed.

File: src/analytics/analytics-on-redshift.ts (hunk 3 above)
Comment (amliuyong) [reply src/analytics/analytics-on-redshift.ts]: removed.

File: src/analytics/analytics-on-redshift.ts (hunk 2 above)
Comment (amliuyong) [reply src/analytics/analytics-on-redshift.ts]: removed

File: src/analytics/analytics-on-redshift.ts (hunk 5 above)
Comment (amliuyong) [reply src/analytics/analytics-on-redshift.ts]: removed

**Pull request merged**